external_modules/total_energy_module/prepare_qe.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+while getopts q: flag
```

Comment:
This script will silently fail and do nothing if `-q ...` is *not* provided. Then it will execute `cd` which is `cd ~` and there do nothing. It may be better to make this a required argument.

Or use `set -u` at the start to fail on undefined variables such as `$root_dir`.

Reply:
That's a very good point I totally overlooked. I fixed this in my last commit.

Reply:
That's not exactly what I meant. I didn't even see that a colon was not used :)

Not using `q:` (without `:`) meant the option did not require an arg, so your test runs of the script should all have failed since the path given to `-q <path>` was never used, strange.

Anyway what I meant was to make this path a required argument (`script.sh foo`) rather than an option (`script.sh -q foo`), i.e.

```sh
#!/bin/sh

set -eu

err(){
    echo "error: $@"
    exit 1
}

[ $# -eq 1 ] || err "need exactly one argument" && root_dir=$1

echo ">>$root_dir<<"
```

If you want to go the option route, then make sure to place any `set ...` *at the top*. In particular `set -u` will make the script fail when `-q <path>` is *not* given since then `$root_dir` is not defined.

```sh
#!/bin/sh

set -eu

err(){
    echo "error: $@"
    exit 1
}

while getopts q: opt; do
    case $opt in
        q) root_dir="${OPTARG}";;
    esac
done

# Since we use set -u (error out on undefined vars, we actually die before this
# check.
[ -z "$root_dir" ] && err "root_dir not defined"

echo ">>$root_dir<<"
```

Reply:
Side note, in the absence of `set -u`, I'd use `[ -z $foo ]` for undefined vars since any undefined shell variable `$foo` is a string of length 0. 

But to not further split hais here, I think the script might do what it should now.

Reply:
Aaaah now I understand what you mean. Yeah there is no reason to do it with a flag in this case, this was done more out of habit. I have edited the scripts to now require a path be provided and fail otherwise, could you have a final look?

Reply:
Looks good.

You can use `grep`'s exit code in the conditional, where succesful `grep` exits 0 (true), else exits 1 (false). Make sure to use single quotes, else the shell will try to expand out `$foxflags` in the script.

```sh
if grep -q '$foxflags -fPIC' install/configure; then
    ....
fi
```

Reply:
I originally had that, but since there is nothing to do if grep exits 0 I instead opted for the comparison I currently have.

Reply:
Ah, sorry then I misread that. 

This would be acomplished by

```sh
if ! grep -q '$foxflags -fPIC'
```

Reply:
Thanks, I tried that but it wouldn't work until I put 

```bash
if ! $(grep -q "\$foxflags -fPIC" install/configure);
```

I have pushed this, I think with this the scripts can be considered finished?